scripts/bin/opx-config-fanout

```diff
@@ -17,173 +17,223 @@ import cps
-    current_speed = fp_obj.get_attr_data('port-speed')
-    if current_mode == _mode and current_speed == _speed:
-        print 'Port already configured as breakout mode %d and speed %d' % (
-              _mode, _speed)
-        sys.exit(1)
 
-    l = []
-
-    npu = fp_obj.get_attr_data('npu-id')
-    phy_port_list = nas_if.physical_ports_for_front_panel_port(fp_obj)
-
-    for phy_port in phy_port_list:
-        if_port = if_cache.get_by_port(
-            npu,
-            phy_port.get_attr_data('port-id'))
-        if if_port is None:
-            continue
-        if_name = if_port.get_attr_data(_g_if_name)
-        print "Deleting.. %s" % if_name
-        cps_obj = cps_object.CPSObject(module = 'dell-base-if-cmn/set-interface',
-                                       data = {'dell-base-if-cmn/set-interface/input/operation': 2,
-                                               _g_if_name: if_name})
-        l.append(('rpc', cps_obj.get()))
-    tr = cps_utils.CPSTransaction(l)
-    tr.commit()
-
-    obj = cps_object.CPSObject(module='base-if-phy/breakout',
-                               data={
-                               'base-if-phy/breakout/input/front-panel-port':
-                                    fp_obj.get_attr_data(
-                                        'base-if-phy/front-panel-port/front-panel-port'),
-                               'base-if-phy/breakout/input/breakout-mode': _mode,
-                               'base-if-phy/breakout/input/port-speed': _speed
-                               })
-    tr = cps_utils.CPSTransaction([('rpc', obj.get())])
-
-    if tr.commit() == False:
-        print "Failed..."
-
-        # revert ports delete
+    for fp in fp_list:
+        fp_obj = nas_fp_cache.get(int(fp))
         l = []
-        # add ports to a transaction
+        npu = fp_obj.get_attr_data('npu-id')
+        phy_port_list = nas_if.physical_ports_for_front_panel_port(fp_obj)
         for phy_port in phy_port_list:
-            ifobj = nas_if.make_interface_from_phy_port(phy_port, current_mode, current_speed)
-            print "Creating interface " + ifobj.get_attr_data(_g_if_name)
-            l.append(('rpc', ifobj.get()))
-
-        # do the actual port create
+            if_port = if_cache.get_by_port(
+                    npu,
+                    phy_port.get_attr_data('port-id'))
+            if if_port is None:
+                continue
+            if_name = if_port.get_attr_data(_g_if_name)
+            print "Deleting.. %s" % if_name
+            cps_obj = cps_object.CPSObject(module = 'dell-base-if-cmn/set-interface',
+                                           data = {'dell-base-if-cmn/set-interface/input/operation': 2,
+                                                   _g_if_name: if_name})
+            l.append(('rpc', cps_obj.get()))
+    
         tr = cps_utils.CPSTransaction(l)
+        tr.commit()
+        print "tr.commit() status"
+        print "done deleting"
```

Comment:
Extraneous print. You need not debug every single action; just keep the user informed about what is happening.